test/tests/functional/pbs_checkpoint.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class TestCheckpoint(TestFunctional):
+
+    """
+    This test suite targets Checkpoint functionality.
+    """
+    def test_checkpoint_abort_with_preemept(self):
+        abort_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        abort_file = os.path.join(pbs_home, 'mom_priv', 'abort_file')
+        localhost = socket.gethostname()
+        fn = self.du.create_temp_file(hostname=localhost, body=abort_script)
+        ret = self.du.run_copy(src=fn,
+                               dest=abort_file, sudo=True, uid='root',
+                               gid='root', mode=0755)
+        if ret['rc'] != 0:
+            self.skipTest('failed to create abort script')
+        c = {'$action': 'checkpoint_abort 30 !' + abort_file + ' %sid'}
+        self.mom.add_config(c)
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+        node_attrs = self.server.status(NODE, id=self.server.shortname)[0]
+        ncpus = node_attrs['resources_available.ncpus']
+        attrs = {'Resource_List.select': '1:ncpus=' + ncpus}
```

Comment:
It's probably easier to just set the number of ncpus at the start of the test than figure out how many ncpus the machine has.